src/java.base/share/classes/java/lang/Shutdown.java

```diff
@@ -157,37 +157,39 @@ static void halt(int status) {
      * which should pass a nonzero status code.
      */
     static void exit(int status) {
-        System.Logger log = getRuntimeExitLogger(); // Locate the logger without holding the lock;
+        logRuntimeExit(status);         // Log without holding the lock;
+
         synchronized (Shutdown.class) {
             /* Synchronize on the class object, causing any other thread
              * that attempts to initiate shutdown to stall indefinitely
              */
-            if (log != null) {
-                Throwable throwable = new Throwable("Runtime.exit(" + status + ")");
-                log.log(System.Logger.Level.DEBUG, "Runtime.exit() called with status: " + status,
-                        throwable);
-            }
             beforeHalt();
             runHooks();
             halt(status);
         }
     }
 
-    /* Locate and return the logger for Shutdown.exit, if it is functional and DEBUG enabled.
-     * Exceptions should not prevent System.exit; the exception is printed and otherwise ignored.
+    /* Locate the logger and log the Runtime.exit(status).
+     * Catch and ignore any and all exceptions.
      */
-    private static System.Logger getRuntimeExitLogger() {
+    private static void logRuntimeExit(int status) {
         try {
             System.Logger log = System.getLogger("java.lang.Runtime");
-            return (log.isLoggable(System.Logger.Level.DEBUG)) ? log : null;
+            if (log.isLoggable(System.Logger.Level.DEBUG)) {
+                Throwable throwable = new Throwable("Runtime.exit(" + status + ")");
+                log.log(System.Logger.Level.DEBUG, "Runtime.exit() called with status: " + status,
+                        throwable);
+            }
         } catch (Throwable throwable) {
-            // Exceptions from locating the Logger are printed but do not prevent exit
-            System.err.println("Runtime.exit() log finder failed with: " + throwable.getMessage());
+            try {
+                // Exceptions from the Logger are printed but do not prevent exit
+                System.err.println("Runtime.exit() logging failed: " + throwable.getMessage());
```

Comment:
Hello Roger, do you think including the exit code in this System.err.println message would be useful too? That would then mean that if the Logger failed for whatever reason, this fallback message would at least include the exit code to provide some hints on what might have triggered the exit.

Reply:
Including the status is simple enough, but has little/nothing to do with why the logging call failed.

Reply:
Agreed - the exit code in the System.err.println fallback would merely be a last resort to provide some possible hint on what code might have triggered an exit.